swarm_copy/tools.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, ClassVar
+
+from openai.lib._tools import pydantic_function_tool
+from pydantic import BaseModel, Field
+
+
+class BaseTool(BaseModel):
+    name: str
+    description: str
```

Comment:
Assuming we wanna continue using this as an Abstract class we should probably go back to using ABC (we thought it caused issues but the problem was somewhere else)

Also, we should add ClassVar to both the name and description